F/FMM3D/build_tarballs.jl

```diff
@@ -0,0 +1,92 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+
+# author: Travis Askham (askhamwhat@gmail.com)
+#
+#TODO: there are a number of stumbling points for the
+# FAST_KER=ON option when cross-compiling. it is completely
+# disabled for now. Some issues:
+# - on musl, darwin, and freebsd various standard libraries
+#   are not found
+# - on mingw there is an invalid regiser error on x86_64 arch
+# - generally, these seem to require a later version of gcc
+#   but we would like to compile for various libgfortran
+#   major versions
+
+using BinaryBuilder, Pkg
+
+name = "FMM3D"
+version = v"0.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/flatironinstitute/FMM3D.git", "2fd129984e48abe437fddd12013dc434563a773a")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd FMM3D/
+
+touch make.inc
+
+# clang has some issue with openmp (lomp not found)
+# force gcc
+if [[ ${target} = *apple* || ${target} = *freebsd* ]]; then
+  export CC="gcc"
+fi
+
+echo "CC=${CC}" >> make.inc;
+echo "CXX=${CXX}" >> make.inc;
+echo "FC=${FC}" >> make.inc;
+
+if [[ ${target} = *mingw* ]]; then
+    echo "FFLAGS= -fPIC -O3 -fno-asynchronous-unwind-tables -funroll-loops" >> make.inc;
+else
+    echo "FFLAGS= -fPIC -O3 -funroll-loops" >> make.inc;
+fi
+
+export OMPFLAGS="-fopenmp"
+export OMPLIBS="-lgomp"
+echo "OMPFLAGS= ${OMPFLAGS}" >> make.inc;
+echo "OMPLIBS= ${OMPLIBS}" >> make.inc;
+
+export LIBS="-lm -lgfortran"
+echo "LIBS = ${LIBS}" >> make.inc;
+
+make -j${nproc} lib OMP=ON
+
+echo "${TARGET}"
+
+# build the correct type of dynamic library for
+# current target
+
+export SHAREFLAGS="-shared -fPIC"
+
+cd lib-static
+ar x libfmm3d.a
+${CC} ${SHAREFLAGS} ${OMPFLAGS} *.o -o "${libdir}/libfmm3d.${dlext}" ${LIBS} ${OMPLIBS}
+cd ..
+
+install_license ${WORKSPACE}/srcdir/FMM3D/LICENSE
+rm make.inc
+exit
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+platforms = expand_gfortran_versions(platforms)
+platforms = expand_cxxstring_abis(platforms)
```

Comment:
I don't think expanding the C++ string ABI is necessary
```suggestion
platforms = expand_gfortran_versions(supported_platforms())
```
I don't get any warning about it when building for `x86_64-linux-gnu`, the resulting library doesn't link to libstc++, and looking for `abi:cxx1` with
```
nm -D ./libfmm3d.so | c++filt |grep 'abi:cxx11'
```
in the library built here for `x86_64-linux-gnu-cxx11` doesn't find any symbol with the special tag